Indicators/IndicatorBase.cs

```diff
@@ -184,7 +185,13 @@ public override bool Equals(object obj)
             // solely relying on reference semantics (think hashset/dictionary impls)
 
             if (ReferenceEquals(obj, null)) return false;
-            if (obj.GetType().IsSubclassOf(typeof (IndicatorBase<>))) return ReferenceEquals(this, obj);
+            var type = obj.GetType();   
+            if (type.IsSubclassOf(typeof(IndicatorBase<IndicatorDataPoint>))||
+                type.IsSubclassOf(typeof(IndicatorBase<IBaseDataBar>)) ||
+                type.IsSubclassOf(typeof(IndicatorBase<TradeBar>)))
```

Comment:
Nice!
Wonder if it would be better to check for the generic interface, to support any `IndicatorBase<T>`, maybe something like https://stackoverflow.com/a/18233467?

Reply:
TBH, I'm not sure how to implement it for the generic `T`, that was the cause of the issue. 

Also, I'm not sue if it worth, this set of generics was taken from the `CommonIndicatorTests`, that is used for *all* indicators.
https://github.com/QuantConnect/Lean/blob/b8b89ab0b60e05dae10c6072e7f9ea1313b378de/Tests/Indicators/CommonIndicatorTests.cs#L49

On top of that, the test added covers more than 40 indicators.

Reply:
Actually something like this https://stackoverflow.com/a/457708 is what we would need, sorry for sending the wrong link, `IndicatorBase<T>` is a class, not an interface 🤦‍♂. I did a quick test and it seems to work correctly

Reply:
Done!